libs/event-manager/web-client/features/list-events/src/lib/list-events.component.ts

```diff
@@ -246,17 +282,27 @@ export class ListEventsComponent {
             first(),
             concatMap((connection) => {
               if (connection === null) {
-                this._matSnackBar.open('Connection missing');
+                this._matSnackBar.open('Connection missing', 'close', {
+                  duration: 5000,
+                });
                 return EMPTY;
               }
 
               return defer(() =>
                 from(connection.confirmTransaction(signature))
               ).pipe(
-                catchError((error) => {
-                  this._matSnackBar.open(error.msg);
-                  return EMPTY;
-                })
+                concatMap(() =>
+                  this._eventApiService
+                    .updateSold(eventFId, ticketQuantity)
```

Comment:
Buy ticket should already update the sold tickets, this should be an ACID Transaction. [https://firebase.google.com/docs/firestore/manage-data/transactions](Firebase transactions docs)